File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/StatedElementUpdatedMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait StatedElementUpdatedMessage extends MessageDeserializer {
  val messageId = 5709

  protected trait Message extends BaseMessage with Serializable {
    val messageId = StatedElementUpdatedMessage.messageId

    val statedElement: StatedElement.TypeImpl

    def serialize(buf: Buffer) {
      statedElement.serialize(buf)
    }

  }

  protected def create(statedElement: StatedElement.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val statedElement = StatedElement.deserialize(buf)

    create(statedElement)
  }

}

object StatedElementUpdatedMessage extends StatedElementUpdatedMessage {
  case class MessageImpl(statedElement: StatedElement.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(statedElement: StatedElement.TypeImpl) = MessageImpl(statedElement)

  def apply(statedElement: StatedElement.TypeImpl) = create(statedElement)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/StatedElementUpdatedMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class StatedElementUpdatedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5709;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public StatedElement statedElement;
    
    public StatedElementUpdatedMessage() { }
    
    public StatedElementUpdatedMessage(StatedElement statedElement) {
        this.statedElement = statedElement;
    }
    
    @Override
    public void serialize(Buffer buf) {
        statedElement.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        statedElement = new StatedElement();
        statedElement.deserialize(buf);
    }
    
}
